sql_assesment.txt

```diff
@@ -0,0 +1,1786 @@
+| 2001 | Hoffman  | London  |    100 | 1001 |
+| 2002 | Giovanni | Rome    |    200 | 1003 |
+| 2003 | Liu      | Sanjose |    200 | 1002 |
+| 2004 | Grass    | Berlin  |    300 | 1002 |
+| 2006 | Ciemens  | London  |    100 | 1001 |
+| 2008 | Cisneros | SanJose |    300 | 1007 |
+| 2007 | Periera  | Rome    |    100 | 1004 |
++------+----------+---------+--------+------+
+
+
+
+
+10) Match salespeople to customers according to what city they live in. 
+
+select s.sname,c.cname, s.city from salespeople as s inner join customers as c on s.city= c.city;
+
++--------+----------+---------+
+| sname  | cname    | city    |
++--------+----------+---------+
+| Fran   | Hoffman  | London  |
+| Motika | Hoffman  | London  |
+| Peel   | Hoffman  | London  |
+| Serres | Liu      | SanJose |
+| Fran   | Ciemens  | London  |
+| Motika | Ciemens  | London  |
+| Peel   | Ciemens  | London  |
+| Serres | Cisneros | SanJose |
++--------+----------+---------+
+
+
+
+
+11) Find all the customers in SanJose who have a rating above 200. 
+
+select * from customers where rating>200 and city="Sanjose";
+
++------+----------+---------+--------+------+
+| CNUM | CNAME    | CITY    | RATING | SNUM |
++------+----------+---------+--------+------+
+| 2008 | Cisneros | SanJose |    300 | 1007 |
++------+----------+---------+--------+------+
+
+
+
+
+
+12) List the names and commissions of all salespeople in London. 
+
+select sname, comm from salespeople where city = "LONDON";
+
++--------+------+
+| sname  | comm |
++--------+------+
+| Peel   |   12 |
+| Motika |   11 |
+| Fran   |   25 |
++--------+------+
+
+
+
+13) List all the orders of Salesperson Motika from the orders table.
+
+select * from orders where snum in (select snum from salespeople where sname= "Motika");
+
++------+--------+------------+------+------+
+| ONUM | AMT    | ODATE      | CNUM | SNUM |
++------+--------+------------+------+------+
+| 3002 | 1990.1 | 1990-03-10 | 2007 | 1004 |
++------+--------+------------+------+------+
+
+
+
+14) Find all customers who booked orders on October 3. 
+
+
+select distinct(c.cname) from customers as c inner join orders o on c.cnum=o.cnum and o.odate= "1990-03-10";
+
+
++----------+
+| cname    |
++----------+
+| Hoffman  |
+| Liu      |
+| Cisneros |
+| Periera  |
++----------+
+
+
+
+
+15) Give the sums of the amounts from the Orders table, grouped by date, eliminating all those dates where the SUM was not at least 2000 above the maximum Amount. 
+
+select odate,max(amt),sum(amt) from orders group by odate having (sum(amt)-(max(amt))>2000);
+
++------------+----------+----------+
+| odate      | max(amt) | sum(amt) |
++------------+----------+----------+
+| 1990-03-10 |  5160.45 |  8944.59 |
++------------+----------+----------+
+
+
+
+
+16) Select all orders that had amounts that were greater than at least one of the orders from October 6. 
+
+select * from orders where amt> any(select amt from orders where odate= "1990-06-10");
+
```

Comment:
Can we optimize this query? ANY keeps checking unless it doesn't find a greater order.